test/problem3_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rspec'
+require_relative '../src/problem3'
+
+describe Problem3 do
```

Comment:
Yes, the test fails, I will try to come up with a better implementation for this problem.